### PR TITLE
Clean the Social Networks page

### PR DESCRIPTION
**Pull request by RicardoVaz1**
Why:
* To make the code cleaner

How:
* Fixing the Social Network page and component

### CODE REVIEW

File: frontend/src/components/SocialMediaButton/index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import "./index.css";
 
+import Default from "../../assets/imgs/socialButtons/default.png";
```
Comment (mirhamasala) [inline frontend/src/components/SocialMediaButton/index.js:3]: Why do you need this?

File: frontend/src/components/SocialMediaButton/index.js [hunk 2]
```diff
@@ -9,60 +10,35 @@ import Dribbble from "../../assets/imgs/socialButtons/dribble.png";
...
-    link = links.dribbble + username;
-  }
-  if (name === "medium") {
-    logo = Medium;
-    link = links.medium + username;
-  }
-  if (name === "twitter") {
-    logo = Twitter;
-    link = links.twitter + username;
+  const info = {
```
Comment (mirhamasala) [inline frontend/src/components/SocialMediaButton/index.js:16]: I suggest thinking of a better name than `info`. Try to be descriptive.

File: frontend/src/components/SocialMediaButton/index.js [hunk 3]
```diff
@@ -9,60 +10,35 @@ import Dribbble from "../../assets/imgs/socialButtons/dribble.png";
...
+  if (name === "github") { i = 1 }
+  else if (name === "linkedin") { i = 2 }
+  else if (name === "behance") { i = 3 }
+  else if (name === "facebook") { i = 4 }
+  else if (name === "instagram") { i = 5 }
+  else if (name === "dribbble") { i = 6 }
+  else if (name === "medium") { i = 7 }
+  else if (name === "twitter") { i = 8 }
+  else { i = 0 }
+
```
Comment (mirhamasala) [inline frontend/src/components/SocialMediaButton/index.js:37]: Still not functional.

**Comment by mirhamasala** [comment]
I would add that this isn't a `button` but a `link`

File: frontend/src/components/SocialMediaButton/index.js [hunk 4]
```diff
@@ -9,60 +10,35 @@ import Dribbble from "../../assets/imgs/socialButtons/dribble.png";
...
   return (
     <div>
```
Comment (mirhamasala) [inline frontend/src/components/SocialMediaButton/index.js:39]: Empty `<div>`

File: frontend/src/components/SocialMediaButton/index.js (hunk 2 above)
Comment (mirhamasala) [reply frontend/src/components/SocialMediaButton/index.js:16]: Could you use the social networks name as a key and have the `url` as value?

File: frontend/src/components/SocialMediaButton/index.js (hunk 1 above)
Comment (RicardoVaz1) [reply frontend/src/components/SocialMediaButton/index.js:3]: The idea was if the prop name was empty, it should show a default image. Should I remove it?

File: frontend/src/components/SocialMediaButton/index.js (hunk 2 above)
Comment (RicardoVaz1) [reply frontend/src/components/SocialMediaButton/index.js:16]:
```
const socialMedias = {
    "": { url: "#", logo: Default },
    "github": { url: `https://github.com/${username}`, logo: Github },
    "linkedin": { url: `https://www.linkedin.com/company/${username}`, logo: LinkedIn },
    ...
  }
```
Should it be something like this?

File: frontend/src/components/SocialMediaButton/index.js (hunk 1 above)
Comment (mirhamasala) [reply frontend/src/components/SocialMediaButton/index.js:3]: Can the `name` be empty?

File: frontend/src/components/SocialMediaButton/index.js (hunk 2 above)
Comment (mirhamasala) [reply frontend/src/components/SocialMediaButton/index.js:16]: I don't think `"": { url: "#", logo: Default },` should be part of this object. If needed, you can return that in an `else` in the function that you're still to create.

Comment (mirhamasala) [reply frontend/src/components/SocialMediaButton/index.js:16]: Also,  you could also use the key for `logo`. That way, you wouldn't need `url` and `logo` but just `"github": https://github.com/${username}` - right?

File: frontend/src/pages/Signup.js [hunk 5]
```diff
@@ -16,11 +16,11 @@ const Signup = () => {
...
-        <SocialMediaButton name="linkedin" username="wearesubvisual" />
-        <SocialMediaButton name="behance" username="subvisual" />
-        {/* <SocialMediaButton name="facebook" username="subvisual.co" />
-        <SocialMediaButton name="instagram" username="wearesubvisual" /> */}
+        <SocialMediaLink name="github" username="subvisual" />
+        <SocialMediaLink name="linkedin" username="wearesubvisual" />
+        <SocialMediaLink name="behance" username="subvisual" />
+        {/* <SocialMediaLink name="facebook" username="subvisual.co" />
+        <SocialMediaLink name="instagram" username="wearesubvisual" /> */}
       </div>
```
Comment (mirhamasala) [inline frontend/src/pages/Signup.js:24]: You may want to think about who should store the data object. This page or `SocialMediaLink`

File: frontend/src/components/SocialMediaLink/index.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    "": { url: "#", logo: Default },
+    github: { url: `https://github.com/${username}`, logo: Github },
+    linkedin: { url: `https://www.linkedin.com/company/${username}`, logo: Linkedin },
+    behance: { url: `https://www.behance.net/${username}`, logo: Behance },
+    facebook: { url: `https://www.facebook.com/${username}`, logo: Facebook },
+    instagram: { url: `https://www.instagram.com/${username}`, logo: Instagram },
+    dribbble: { url: `https://dribbble.com/${username}`, logo: Dribbble },
+    medium: { url: `https://medium.com/${username}`, logo: Medium },
+    twitter: { url: `https://twitter.com/${username}`, logo: Twitter },
+  }
```
Comment (mirhamasala) [inline frontend/src/components/SocialMediaLink/index.js:24]: I'm not sure this data should "live" in this component either. We could think of extracting it into a `constants.js` file. Then you could import the data you need in the parent components, and pass the props down. What do you think @joaojusto ?

Comment (mirhamasala) [reply frontend/src/components/SocialMediaLink/index.js]: @joaojusto Nevermind. They're planning to move all of this to the database.

**Comment by mirhamasala** [comment]
@RicardoVaz1 I converted this to a DRAFT PR since it includes test code.

**Comment by RicardoVaz1** [comment]
@mirhamasala I fixed the merge conflicts

**Comment by mirhamasala** [comment]
> @mirhamasala I fixed the merge conflicts

Needs to be rebased, still.

**Comment by mirhamasala** [comment]
Closed in favor of #38

**Pull request closed without merging**